drivers/display/display_lpm013m126.c

```diff
@@ -0,0 +1,257 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2025 Silicon Signals Pvt. Ltd.
+ * Author: Bhavin Sharma <bhavin.sharma@siliconsignals.io>
+ * Author: Elgin Perumbilly <elgin.perumbilly@siliconsignals.io>
+ */
+
+#define DT_DRV_COMPAT jdi_lpm013m126
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(lpm013m126, CONFIG_DISPLAY_LOG_LEVEL);
+
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/kernel.h>
+
+/* Panel properties */
+#define LPM_WIDTH       DT_INST_PROP(0, width)
+#define LPM_HEIGHT      DT_INST_PROP(0, height)
+#define LPM_BPP         3
+
+/* Command bytes */
+#define LPM_WRITELINE_CMD   0x80
+#define LPM_ALLCLEAR_CMD    0x20
+
+/* bit-reversal of row address */
+static inline uint8_t bitrev8(uint8_t x)
+{
+	x = ((x & 0xF0) >> 4) | ((x & 0x0F) << 4);
+	x = ((x & 0xCC) >> 2) | ((x & 0x33) << 2);
+	x = ((x & 0xAA) >> 1) | ((x & 0x55) << 1);
+	return x;
+}
+
+/* The native format (1 bit per channel) is rather unusual. LVGL and
+ * other libraries don't support it. In addition, the format is not
+ * very convenient for the application. So, we prefer to advertise
+ * a well known format and convert it under the hood.
+ * A native implementation of this format would allow to save memory
+ * for the frame buffer (11kB instead of 30kB).
+ */
+static inline uint8_t rgb565_to_rgb3(uint16_t color)
+{
+	uint8_t r = FIELD_GET(0xF800, color);
+	uint8_t g = FIELD_GET(0x07E0, color);
+	uint8_t b = FIELD_GET(0x001F, color);
+
+	r >>= 4;
+	g >>= 5;
+	b >>= 4;
+	return (r << 2) | (g << 1) | b;
+}
+struct lpm013m126_data {
+	struct k_timer vcom_timer;
+	int vcom_state;
+};
+
+struct lpm013m126_config {
+	struct spi_dt_spec bus;
+	struct gpio_dt_spec disp_gpio;
+	struct gpio_dt_spec extcomin_gpio;
+	uint32_t extcomin_freq;
+};
```

Comment:
Usually, we tend to declare these struct before any functions (sorry, I missed that in the previous iterations).